app/controllers/miq_task_controller.rb

```diff
@@ -128,7 +128,16 @@ def cancel_task
   # Delete selected tasks
   def delete_tasks
     assert_privileges("miq_task_delete")
-    delete_tasks_from_table(find_checked_items, "Delete selected tasks")
+
+    checked_items = find_checked_items
+    new_checked_items = []
+    checked_items.each do |item|
+      if params["select-row-#{item}"] == "on"
+        new_checked_items.push(item)
+      end
+    end
```

Comment:
```suggestion
    new_checked_items = find_checked_items.select { |item| params["select-row-#{item}"] == "on" }
```

Reply:
I'll be honest I don't understand what this code is saying. Do only newly checked items have the "select-row-" on?

Reply:
The delete button doesn't instantly delete the tasks, it instead queues their deletion. So what could happen is you click delete on an item and when the page reloads the item is still in the table checked off so it looks like the delete button did nothing.